### ISSUE TITLE
Transferring leadership should have a confirmation popup

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitica#11767]
**Qasmi:** Party Leadership is instantly transferred to another member without confirmation when you click on "Transfer Ownership>3 dots button>View Party". It should ask for confirmation before transferring as sometimes while removing a member one can accidently make them the leader.
**shanaqui:** Hmm. Sabe, is there meant to be a confirmation? Should I put this on Github, you think?
**SabreCat:** Yeah, that does seem like the sort of process that deserves an are-you-sure!

I've confirmed the behaviour Qasmi saw happens consistently on web, there was no accidentally clicked through confirmation or anything.

**Comment by SabreCat** [comment]
A Javascript `confirm()` prompt would be fine here, though if you want extra contributor-tier brownie points, you could abstract https://github.com/HabitRPG/habitica/blob/develop/website/client/src/components/payments/cancelModalConfirm.vue to a more generic mixin and employ that!

**Comment by prachi15091999** [comment]
Sir
My team of some students are looking for an open source project to contribute to, and this seems like a good starting point to work on.
We have thought to solve this issue and some other issues of this project that could be  added to this proejct.
So our work considered for merging with the project?

**Comment by prachi15091999** [comment]
Sir
My team of some students are looking for an open source project to contribute to, and this seems like a good starting point to work on.
We have thought to solve this issue and some other issues of this project that could be added to this proejct.
We will make a pop up dialogue box to solve this issue with good design.
So our work considered for merging with the project?

**Comment by Alys** [comment]
@prachi15091999 Hi! Thank you for wanting to work on this! I've marked it as in progress for you.

If you'd like to do something more complex than a Javascript confirm() prompt, it may be best to post some details here first about what you are thinking of doing. You're also welcome to post here if you have any questions or run into problems!

By the way, just so you know, not everyone here is male and some of us don't identify as female either so it's best to avoid gendered terms like "Sir" or "Madam". :)

**Comment by prachi15091999** [comment]
Hello everyone
my team wrote the code for the popup that was placed in the transfer ownership function but we are facing issues in hosting habitica locally.The error when all our members try to host the site locally is as follows:-
1.The ''vue-cli-service'' command was not found .
2. The ''index.html'' was not found .
The screenshots of error that is shown to us is attached below.
These errors arise may be due to the different versions in your directories from our server therefore can you provide all the versions of directories we have to install for hosting it locally
We shall be highly thankful if you either assist us with these error or give us personal guidance for the same . 
![WhatsApp Image 2020-09-11 at 3 31 11 PM](https://user-images.githubusercontent.com/44581727/92921492-c00fa300-f451-11ea-9326-cacd321c22a9.jpeg)
![WhatsApp Image 2020-09-11 at 3 31 12 PM (1)](https://user-images.githubusercontent.com/44581727/92921509-c736b100-f451-11ea-90bf-b971ba1f64d1.jpeg)
![WhatsApp Image 2020-09-11 at 3 31 12 PM (2)](https://user-images.githubusercontent.com/44581727/92921531-cf8eec00-f451-11ea-8661-7f7c08beb9a7.jpeg)

**Comment by paglias** [comment]
@prachi15091999 could you try running `npm install` from `website/client` and see if it works after that?

**Comment by shanaqui** [comment]
Hi @prachi15091999! Are you still working on this issue? If not, or if you don't reply in a week, I'll mark it as help wanted again just to make sure it doesn't go stale. :)

**Comment by prachi15091999** [comment]
Yes we are working on the issue sorry for the delay caused due to  our exams.

**Comment by prachi15091999** [comment]
Yeah, we are working on this matter, and we apologize for the time taken, we got busy due to the academic activities. Also now we are able to host the habitica locally.We will provide you with the code as soon it's completed.

**Comment by prachi15091999** [comment]
Yeah, we are working on this issue, and we apologize for the time taken, we got busy due to the academic activities. Also now we are able to host the habitica locally.We will provide you with the code as soon it's completed.

**Comment by shanaqui** [comment]
No worries! Just checking in. :)

**Comment by prachi15091999** [comment]
My team have some issues while resolving the issue. Currently, we have developed a modal similar to the removeMemberModal.vue and then implemented in the memberModal.vue in the promotetoleader function like you implement in the remove member function,but the code dosen't work our modal does not called by memberModal.Then we import our modal in that as you implement the removeMemberModal, and still our Modal does not called by Member Modal.Can you make few suggestion that helps us in getting  the correct way in importing our modal ?

**Comment by paglias** [comment]
@prachi15091999 without seeing the code it's difficult to understand the problem, if you push a PR I'll give it a look

**Comment by shanaqui** [comment]
@prachi15091999 Hi there! Just checking in whether you're still interested in working on this issue. 

If we don't hear from you within a week, or you're not interested anymore, I'll mark this as help wanted again. :) No rush to complete a fix or anything, we're just making sure the issue doesn't go stale!

**Comment by prachi15091999** [comment]
Yes ,We are working on it and we complete our work  by next week.

**Comment by prachi15091999** [comment]
We have done our code for resolving the issue and implement the popup succesfully .Soon you will get the pull request form our side with the github username 18ucs077.
The popup we have added is shown in the attached image .In this popup the  username and avatar of the member to whom the leadership should be transferred is shown.

![WhatsApp Image 2020-11-30 at 9 53 29 PM](https://user-images.githubusercontent.com/44581727/100637545-055cb600-3359-11eb-8186-94c2e81c6ca0.jpeg)

**Comment by akshitagoyal22** [comment]
@shanaqui @paglias Take a look at this PR. I think I have fixed this issue.

**Comment by prachi15091999** [comment]
Our team made a pull request  to fix this issue ..All the tests should be passed when we made request.Take a look at this and accept it if it looks fine to you and if you want some changes please let us know.The pull request is made by 18ucs077
